pallets/gear/src/lib.rs

```diff
@@ -1143,12 +1135,12 @@ pub mod pallet {
                 schedule.limits.table_number.into(),
             )?;
 
-            let code_and_id = CodeAndId::from_parts_unchecked(code, code_id);
-            let code_and_id = InstrumentedCodeAndId::from(code_and_id);
-            T::CodeStorage::update_code(code_and_id.clone());
-            let (code, _) = code_and_id.into_parts();
+            let (instrumented_code, _, metadata) = code.into_parts();
+
+            T::CodeStorage::update_instrumented_code(code_id, instrumented_code.clone());
+            T::CodeStorage::update_code_metadata(code_id, metadata);
```

Comment:
Functions should be merged like `add_code` and `remove_code`

Reply:
Agreed, made a function in storage that writes both of these values

Reply:
and didn't use it*

Reply:
The `update_instrumented_code_and_metadata` function is definitely used in `reinstrument_code` and in the `state_rpc_calls_trigger_reinstrumentation` test